native/src/snappyclient/cpp/impl/ControlConnection.cpp

```diff
@@ -440,3 +449,35 @@ void ControlConnection::failoverExhausted(
   snappyEx.__set_serverInfo(failedServerString);
   throw snappyEx;
 }
+
+void ControlConnection::getConnectedHost(thrift::HostAddress& hostAddr,
+    thrift::HostAddress& connectedHost) {
+  boost::lock_guard<boost::mutex> controlConnLock(this->m_lock);
+
+  auto it = m_controlHostSet.find(hostAddr);
+  if (it != m_controlHostSet.end()){
+    connectedHost = it->second;
+    return;
```

Comment:
Here just use a set as before and its iterator will give the key in the map as required.

Reply:
Modified as per review comment